WooCommerce/Classes/ViewRelated/ReusableViews/LabeledTextViewTableViewCell.swift

```diff
@@ -45,26 +46,29 @@ private extension LabeledTextViewTableViewCell {
         productTextField.backgroundColor = .systemColor(.secondarySystemGroupedBackground)
     }
 
-
     func configureLabelStyle() {
         productStatusLabel.font = UIFont.preferredFont(forTextStyle: .caption1)
         productStatusLabel.textAlignment = .center
-        productStatusLabel.textColor = UIColor.black
-        productLabelHolder.backgroundColor = .gray(.shade5)
-        productLabelHolder.layer.cornerRadius = CGFloat(4.0)
+        productStatusLabel.textColor = BadgeStyle.Colors.textColor
+        productStatusBadgeBg.backgroundColor = BadgeStyle.Colors.defaultBg
+        productStatusBadgeBg.layer.cornerRadius = BadgeStyle.cornerRadius
     }
 
     func configureProductStatusLabel(productStatus: ProductStatus) {
-        if productStatus == ProductStatus.draft {
-                let statusLabel = NSLocalizedString("Draft", comment: "Display label for the product's draft status")
-                productLabelHolder.isHidden = false
-                productStatusLabel.isHidden = false
-                productStatusLabel.text? = statusLabel
-            } else {
-                productLabelHolder.isHidden = true
-                productStatusLabel.isHidden = true
-            }
+        productStatusLabel.text = productStatus.description
+        productStatusBadgeBg.backgroundColor = productStatus == .pending ? BadgeStyle.Colors.pendingBg : BadgeStyle.Colors.defaultBg
+        productStatusBadgeHolder.isHidden = productStatus == .published
```

Comment:
My understanding based on the designs is that we only want the badge to appear for draft and published products. If that's accurate, we also need to hide the badge for products with `privateStatus` and `custom` statuses. (Right now the badge appears for those other statuses.)

And just an idea: Although it's less concise, maybe it would be more readable to use a switch based on `ProductStatus` for this configuration? That way it's clearer how each case is being handled.

Reply:
You mean "draft" and "pending"? Default "published" status was (and still) hidden on both platforms.
I think discussion on Android landed with agreement to display at least private status: https://github.com/woocommerce/woocommerce-android/pull/6369#discussion_r858200115.

I've added "custom" option with a thought that any non-published status will be rarely used and displayed on a top level, but therefore it's more valuable to notice when editing a product.

Agree about `switch` suggestion in the code!

@atorresveiga Hey! Can you tell if you support custom statuses for products and do you display them in a badge on product details?

Reply:
Oops, yep, I meant "draft" and "pending". (Too many statuses starting with "p" and I wrote the wrong thing.)

I missed that discussion about the private status, and I agree with your reasoning about custom statuses as long as we're matching the behavior on both platforms. 👍

Reply:
As far as I understand, Android app works in [a same way](https://github.com/woocommerce/woocommerce-android/blob/69c1f2b85a2cf3ccc02f3d62bf04e04c46d16121/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailCardBuilder.kt#L658-L666) - custom status will be displayed.

So I've just replaced ternary operator with `switch` in c311f4f for clarity.